src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -815,18 +816,14 @@ namespace ScatterKernelDetail{
 
 
       template < typename T_Obj>
-      static void* initHeap(const T_Obj& obj, void* pool, size_t memsize){
-        T_Obj* heap;
-        MALLOCMC_CUDA_CHECKED_CALL(cudaGetSymbolAddress((void**)&heap,obj));
+      static void* initHeap(T_Obj* heap, void* pool, size_t memsize){
```

Comment:
Why not make this functions non-static? The instance could save the `heap` in `initHeap` and reset it in `finalizeHeap`
Then further calls do not need to pass the heap pointer